File: equipo/Java/Regular Expression Matching.java
class Solution {
    static Boolean repetidor(Integer i, Integer j,String s, String p){
            if(i>=s.length() && j>=p.length()){
                return true;
            }
            if(j>=p.length()){
                return false;
            }
            Boolean coinciden = i<s.length() && (s.charAt(i)==p.charAt(j)|| p.charAt(j)=='.');
            if((j+1)< p.length() && p.charAt(j+1)=='*'){
                return repetidor(i,j+2,s,p) || (coinciden && repetidor(i+1,j,s,p));
            }
            if(coinciden){
                return repetidor(i+1,j+1,s,p);
            }
            return false;
        }
    static Boolean isMatch(String s, String p) {        
        return repetidor(0,0,s,p);
    }   
}

File: equipo/python/Regular_Expression_Matching.py
class Solution:
    def isMatch(self, s: str, p: str) -> bool:
        stack = [(0, 0)]
        while stack:
            i, j = stack.pop()
            if j == len(p):
                return i == len(s)
            if j + 1 == len(p) or p[j + 1] != '*':
                print(i,j)
                if i < len(s) and (p[j] == s[i] or p[j] == '.'):
                    stack.append((i + 1, j + 1))
            else:
                stack.append((i, j + 2))
                while i < len(s) and (p[j] == s[i] or p[j] == '.'):
                    stack.append((i + 1, j + 2))
                    i += 1
        return False


    # class Solution:
    #def isMatch(self, s: str, p: str) -> bool:
    #    return bool(re.search(f'^{p}$', s))